src/js/messaging/components/MessageAttachmentsViewItem.jsx

```diff
@@ -1,11 +1,52 @@
 import React from 'react';
+import classNames from 'classnames';
+
+import { authFetch } from '../utils/helpers';
 
 class MessageAttachmentsViewItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.downloadAttachment = this.downloadAttachment.bind(this);
+    this.downloadUrl = null;
+    this.state = { downloading: false };
+  }
+
+  downloadAttachment(event) {
+    event.preventDefault();
+
+    if (this.state.downloading) return;
+
+    if (this.downloadUrl) {
+      window.open(this.downloadUrl, '_blank');
+      return;
+    }
+
+    this.setState({ downloading: true });
+    const requestUrl = this.props.url;
+    authFetch(requestUrl)
+    .then(response => response.blob())
+    .then(blob => {
+      const downloadUrl = URL.createObjectURL(blob);
```

Comment:
Looking further into our config.js, the limit we've set on attachments is 3 MB anyway, so seems like there won't be a problem with the blob size limit.